memory/iter.go

```diff
@@ -52,10 +52,6 @@ func newPrefixIter(ctx context.Context, db *Datastore, prefix []byte, reverse bo
 		pIter.Seek(prefix)
 	}
 
-	// load first item. Seek will also load, so this may be a duplicate
-	// action, but this is idempotent so its OK.
-	pIter.loadLatestItem()
```

Comment:
question: I don't understand what it was doing before nor do I understand why removing it is ok haha.

Reply:
I don't either sorry, and the comment suggested that the original author didn't know too lol

Reply:
Isn't it better to leave it there then in case it is taking care of some edge case we aren't aware of

Reply:
I think it is more expensive to leave it in.  I changed the behaviour of the function called, and as I'm not sure why this is here, I do not know if I want to change the behaviour of this call site.

I know removing the call completely is changing that behaviour, but at least it makes sense, and looks like it makes sense, and can be easily and deliberately changed should we later find that it was (in some form) needed.

Reply:
let me investigate for a moment

Reply:
Looks fine to delete it. I think I was being overly cautious during this implementation 🙃